test/python/transpiler/test_template_matching.py

```diff
@@ -647,6 +655,26 @@ def test_naming_clash_multiparameter(self):
 
         self.assertEqual(circuit_out, expected)
 
+    def test_clifford_templates(self):
+        """Tests TemplateOptimization pass on several larger examples."""
+        template_list = [
+            clifford_2_1(),
+            clifford_2_2(),
+            clifford_2_3(),
+            clifford_2_4(),
+            clifford_3_1(),
+        ]
+        pm = PassManager(TemplateOptimization(template_list=template_list))
+        for seed in range(10):
+            qc = random_clifford_circuit(
+                num_qubits=5,
+                num_gates=100,
+                gates=["x", "y", "z", "h", "s", "sdg", "cx", "cz", "swap"],
+                seed=seed,
+            )
+            qc_opt = pm.run(qc)
+            self.assertTrue(Operator(qc) == Operator(qc_opt))
```

Comment:
Rather than using random circuits here (especially because we're importing the function from miles away and need to override some bits of it to prevent it failing), it might be nicer to use a test circuit that has at least a few of the templates in that we're explicitly testing.  The templates are pretty short, so we probably don't need huge test cases.

Reply:
Thanks. I have an extra test which might or might not be what you have asked for, but which seemed good to add anyway: the case when one template optimization creates an opportunity for another template optimization. Though this is not directly related to the bug at hand.

I do feel that we want to keep the slightly larger random testcases. The template optimization algorithm is quite involved, and small handmade testcases just don't cover the huge variety of things that potentially might go wrong. Having a check that for a sufficiently large random circuit the optimizations produce an equivalent circuit definitely helps (and this would have caught the bug of forgetting to update the nodes' predecessors).

Reply:
BTW, I have just understood how nice of an output we get from  `coveralls` (I am really positively amazed), however the simple test code coverage might not be sufficient to catch more complex bugs.

Reply:
I have added one more simple testcase (which would have caught the bug), but I am in still in favor of keeping medium-sized clifford testcases.